### PR TITLE
added condition based on custom config

### PR DESCRIPTION
**Pull request by rishkarajgi**
## Description

Loads external vector config from secrets if customConfig flag is enabled

## Related Issues

https://github.com/astronomer/issues/issues/4733

## Testing

Do not merge this PR until this text is replaced with details about how these changes were tested.

## Merging

0.29.2

### CODE REVIEW

**Review by pgvishnuram — APPROVED** [review]
configs are validated

File: tests/chart_tests/test_logging_sidecar.py [hunk 1]
```diff
@@ -42,3 +44,50 @@ def test_logging_sidecar_config_disabled(self, kube_version):
...
+                    component: "${COMPONENT:--}"
+                    workspace: "${WORKSPACE:--}"
+                    release: "${RELEASE:--}"
+              sinks:
+                out:
+                  type: datadog
+                  inputs:
+                    - transform_syslog
+                """
+        )
```
Comment (danielhoherd) [inline tests/chart_tests/test_logging_sidecar.py:81]: Another way we've done this that is more compact is to use json for this. https://github.com/astronomer/astronomer/blob/c66c8f75e46d5d6ec874c5631a92ab3890ed927d/tests/chart_tests/test_ingress.py#L31-L39

File: tests/chart_tests/test_logging_sidecar.py [hunk 2]
```diff
@@ -42,3 +44,50 @@ def test_logging_sidecar_config_disabled(self, kube_version):
...
+
+    def test_logging_sidecar_custom_config(self, kube_version):
+        """Test logging sidecar config with customConfig flag enabled"""
+        custom_side_carconfig = textwrap.dedent(
+            """
+        loggingSidecar:
+          enabled: true
+          customConfig: true
+          name: sidecar-logging-consumer
+          airflowSidecarConfig: |2
```
Comment (danielhoherd) [inline tests/chart_tests/test_logging_sidecar.py:56]: What is this `|2` syntax? I can't find any info on it, and it looks like it behaves the same as `|`

<https://yaml-multiline.info>

File: tests/chart_tests/test_logging_sidecar.py [hunk 3]
```diff
@@ -42,3 +44,50 @@ def test_logging_sidecar_config_disabled(self, kube_version):
             show_only="templates/logging-sidecar-configmap.yaml",
         )
         assert len(docs) == 0
+
+    def test_logging_sidecar_custom_config(self, kube_version):
+        """Test logging sidecar config with customConfig flag enabled"""
+        custom_side_carconfig = textwrap.dedent(
```
Comment (danielhoherd) [inline tests/chart_tests/test_logging_sidecar.py:50]: Seems like this var should be either `custom_sidecar_config` or `custom_side_car_config`

**Review by danielhoherd — CHANGES_REQUESTED** [review]
Looks mostly good. I left a few comments that I'd like to see answers for.

File: tests/chart_tests/test_logging_sidecar.py (hunk 3 above)
Comment (rishkarajgi) [reply tests/chart_tests/test_logging_sidecar.py]: Thanks. Corrected

File: tests/chart_tests/test_logging_sidecar.py (hunk 2 above)
Comment (rishkarajgi) [reply tests/chart_tests/test_logging_sidecar.py]: It denotes the block indentation number.  As you said no change in behaviour

File: tests/chart_tests/test_logging_sidecar.py (hunk 1 above)
Comment (rishkarajgi) [reply tests/chart_tests/test_logging_sidecar.py]: Took inspiration from this - https://github.com/apache/airflow/blob/9e2658ea8850b0bacba8a3fe82c457dbeb48d591/tests/charts/test_extra_configmaps_secrets.py#L32
Agree the above is more compact

**Comment by rishkarajgi** [comment]
> Looks mostly good. I left a few comments that I'd like to see answers for.

@danielhoherd PTAL

File: tests/chart_tests/test_logging_sidecar.py [hunk 4]
```diff
@@ -42,3 +44,50 @@ def test_logging_sidecar_config_disabled(self, kube_version):
...
+                    - generate_syslog
+                  fields:
+                    component: "${COMPONENT:--}"
+                    workspace: "${WORKSPACE:--}"
+                    release: "${RELEASE:--}"
+              sinks:
+                out:
+                  type: datadog
+                  inputs:
+                    - transform_syslog
```
Comment (danielhoherd) [inline tests/chart_tests/test_logging_sidecar.py:79]: The indentation in this block is inconsistent and fails yamllint.
```suggestion
        loggingSidecar:
          enabled: true
          customConfig: true
          name: sidecar-logging-consumer
          airflowSidecarConfig: |
            log_schema:
              timestamp_key : "@timestamp"
            data_dir: "${SIDECAR_LOGS}"
            sources:
              generate_syslog:
                type: file
                include:
                  - "${SIDECAR_LOGS}/*.log"
                read_from: beginning
            transforms:
              transform_syslog:
                type: add_fields
                inputs:
                  - generate_syslog
                fields:
                  component: "${COMPONENT:--}"
                  workspace: "${WORKSPACE:--}"
                  release: "${RELEASE:--}"
            sinks:
              out:
                type: datadog
                inputs:
                  - transform_syslog
```

**Review by danielhoherd — APPROVED** [review]
LGTM other than the yaml whitespace inconsistencies, which aren't a show-stopper.

File: tests/chart_tests/test_logging_sidecar.py (hunk 4 above)
Comment (rishkarajgi) [reply tests/chart_tests/test_logging_sidecar.py]: done

**Pull request merged**